### ISSUE TITLE
Impossible to make http request with escaped colon ':' characters in the path

### ISSUE THREAD

**Issue by elerch** [ziglang/zig#17015]
### Zig Version

0.11.0

### Steps to Reproduce and Observed Behavior

I am working on an AWS SDK for zig. As part of this, I need to pass in function parameters into paths, some of which include colons, most notably the "arn", used to identify a resource in AWS. This is not part of the path, this is a parameter, so it should be escaped (but only once). There is currently no way to do this currently with the standard library.  in `std.http.Client.Request`, the `start()` method uses a format specifier of `{/}`, which is interpreted by `std.Uri` in the format function, which escapes the path. This is great most of the time, but it prevents the use of pre-escaped colons in the path. My choices are either a path without any escaping (since a colon is an unreserved path character), or with double escaping (because I passed in an escaped path and so the '%' character requires it).

For a use case such as mine, I need to use the ':' character as part of a parameter, so it needs to be escaped, but not double escaped. This is not currently possible.

I believe the right answer here is to take the URI verbatim and leave the escaping to the caller (this seems to be how many other frameworks operate). Some parameter on the `request.start()` to control the behavior might be an alternative as well.

Steps to reproduce:

Program:
```zig
const std = @import("std");

pub fn main() !void {
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    defer _ = gpa.deinit();
    const allocator = gpa.allocator();

    const stdout_file = std.io.getStdOut().writer();
    var bw = std.io.bufferedWriter(stdout_file);
    const stdout = bw.writer();

    var client: std.http.Client = .{
        .allocator = allocator,
        .proxy = .{
            .protocol = .plain,
            .host = "localhost",
            .port = 8080,
        },
    };
    defer client.deinit();

    var req = try client.request(.GET, try std.Uri.parse("https://httpbin.org/anything/arn:aws:lambda"), .{ .allocator = allocator }, .{});
    defer req.deinit();
    try req.start();
    try req.wait();

    var response_body = try req.reader().readAllAlloc(allocator, std.math.maxInt(usize));
    defer allocator.free(response_body);
    try stdout.print("{s}\n", .{response_body}); // not escaped at all

    var req_escaped = try client.request(.GET, try std.Uri.parse("https://httpbin.org/anything/arn%3Aaws%3Alambda"), .{ .allocator = allocator }, .{});
    defer req_escaped.deinit();
    try req_escaped.start();
    try req_escaped.wait();

    var response_body_escaped = try req_escaped.reader().readAllAlloc(allocator, std.math.maxInt(usize));
    defer allocator.free(response_body_escaped);
    try stdout.print("{s}\n", .{response_body_escaped}); // double escaped
    //
    try bw.flush();
}
```

Output (note the URLs - first is plain, second is double-escaped): 

```json
{
  "args": {},
  "data": "",
  "files": {},
  "form": {},
  "headers": {
    "Accept-Encoding": "gzip, deflate, zstd",
    "Host": "httpbin.org",
    "User-Agent": "zig/0.11.0 (std.http)",
    "X-Amzn-Trace-Id": "Root=1-64eed38b-1b36fa5b71ad549309306271"
  },
  "json": null,
  "method": "GET",
  "origin": "50.39.111.175",
  "url": "https://httpbin.org/anything/arn:aws:lambda"
}

{
  "args": {},
  "data": "",
  "files": {},
  "form": {},
  "headers": {
    "Accept-Encoding": "gzip, deflate, zstd",
    "Host": "httpbin.org",
    "User-Agent": "zig/0.11.0 (std.http)",
    "X-Amzn-Trace-Id": "Root=1-64eed390-1a130521338835a31cebaf50"
  },
  "json": null,
  "method": "GET",
  "origin": "50.39.111.175",
  "url": "https://httpbin.org/anything/arn%253Aaws%253Alambda"
}
```

### Expected Behavior

I would expect to have the ability to somehow get the standard library to issue a call with single escaped colon characters in the path.

**Comment by truemedian** [comment]
Percent encoding isn't an "escape", it's an encoding applied only to the URI, it should be decoded on the other end, in which case `:` and `%3A` in a URI have the exact same meaning, as both are decoded to `:`. Any server that requires `:` to be percent encoded isn't handling URIs correctly.

**Comment by elerch** [comment]
I used the term "escape" due to it's usage here: https://github.com/ziglang/zig/blob/master/lib/std/Uri.zig#L22

The point is that I, as a user, cannot issue a request that has %3A in a path, it will get encoded (via the escapePath function in Uri) by the standard library as %253A, which is either double encoded or double escaped depending on your terminology.

"Together we serve the users" - regardless of how a service should be interpreting them or the reasons why (I personally believe this particular service has made an appropriate decision for their service, but that's another discussion), it is unlikely that we will change a hugely popular service like AWS Lambda (which is where I tripped across this issue). But I do have this control in many other programming languages, in curl, etc.

It's funny that you're the one responding @truemedian - I just ported my working code from zFetch to use the standard library and remove dependencies, and that's where I bumped into this. :)
